packages/dev/core/src/Lights/IES/iesLoader.ts

```diff
@@ -0,0 +1,202 @@
+    lumensPerLamp?: number;
+    candelaMultiplier?: number;
+    numberOfVerticalAngles: number;
+    numberOfHorizontalAngles: number;
+    horizontalAngles: number[];
+    verticalAngles: number[];
+    photometricType?: number;
+    unitsType?: number;
+    width?: number;
+    length?: number;
+    height?: number;
+    ballastFactor?: number;
+    fileGenerationType?: number;
+    inputWatts?: number;
+    candelaValues: number[][];
+}
+
+interface IDataPointer {
+    lines: string[];
+    index: number;
+}
+
+function lineToArray(line: string): number[] {
+    return line
+        .split(" ")
+        .filter((x) => x !== "")
+        .map((x) => parseFloat(x));
+}
+
+function readArray(dataPointer: IDataPointer, count: number, targetArray: number[]) {
+    while (targetArray.length !== count) {
+        const line = lineToArray(dataPointer.lines[dataPointer.index++]);
+        targetArray.push(...line);
+    }
+}
+
+function interpolateCandelaValues(data: IIESData, phi: number, theta: number): number {
+    let phiIndex = 0;
+    let thetaIndex = 0;
+    let startTheta = 0;
+    let endTheta = 0;
+    let startPhi = 0;
+    let endPhi = 0;
+
+    // Check if the angle is outside the range
+    for (let index = 0; index < data.numberOfHorizontalAngles - 1; index++) {
+        if (theta < data.horizontalAngles[index + 1] || index === data.numberOfHorizontalAngles - 2) {
+            thetaIndex = index;
+            startTheta = data.horizontalAngles[index];
+            endTheta = data.horizontalAngles[index + 1];
+
+            break;
+        }
+    }
+
+    for (let index = 0; index < data.numberOfVerticalAngles - 1; index++) {
+        if (phi < data.verticalAngles[index + 1] || index === data.numberOfVerticalAngles - 2) {
+            phiIndex = index;
+            startPhi = data.verticalAngles[index];
+            endPhi = data.verticalAngles[index + 1];
+
+            break;
+        }
+    }
+
+    const deltaTheta = endTheta - startTheta;
+    const deltaPhi = endPhi - startPhi;
+
+    if (deltaPhi === 0) {
+        return 0;
+    }
+
+    // Interpolate
+    const t1 = deltaTheta === 0 ? 0 : (theta - startTheta) / deltaTheta;
+    const t2 = (phi - startPhi) / deltaPhi;
+
+    const nextThetaIndex = deltaTheta === 0 ? thetaIndex : thetaIndex + 1;
+
+    const v1 = Lerp(data.candelaValues[thetaIndex][phiIndex], data.candelaValues[nextThetaIndex][phiIndex], t1);
+    const v2 = Lerp(data.candelaValues[thetaIndex][phiIndex + 1], data.candelaValues[nextThetaIndex][phiIndex + 1], t1);
+    const v = Lerp(v1, v2, t2);
+
+    return v;
+}
+
+/**
+ * Generates an IES texture from a string representing the IES data.
+ * @param source defines a string representing the IES data
+ * @param engine defines the engine to use
+ * @returns the IES texture
+ * @see https://ieslibrary.com/browse
+ * #UQGPDT
+ */
+export function LoadIESData(source: string, engine: AbstractEngine): RawTexture {
+    // Read data
+    const dataPointer = {

```

Comment:
I think Typescript does not know that `dataPointer` is of type `IDataPointer`, so set it explicitely?